AK/Math.h

```diff
@@ -21,243 +21,54 @@ constexpr T Pi = 3.141592653589793238462643383279502884L;
 template<FloatingPoint T>
 constexpr T E = 2.718281828459045235360287471352662498L;
 
-namespace Details {
-template<size_t>
-constexpr size_t product_even();
-template<>
-constexpr size_t product_even<2>() { return 2; }
-template<size_t value>
-constexpr size_t product_even() { return value * product_even<value - 2>(); }
```

Comment:
How are these architecture specific?

Reply:
I moved them because they are/should only used by the x86 implementation.